### PR TITLE
Implement View Components

### PR DESCRIPTION
**Pull request by ted3x**
Implemented chatActivity. I
mplemented ChatFragment + ChatViewModel. 
Implemented ChatAdapter. 
Added Long extension converting toDate. 
Implemented Constants object. 
Implemented MessageReceiver as BroadcastReceiver. 
Created styles for textviews.

### CODE REVIEW

File: app/src/main/java/com/c0d3in3/chatapp/ui/ChatFragment.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        super.onResume()
+        messageReceiver = MessageReceiver {
+            if (it.messageOwner != this.messageOwner) viewModel.addNewMessage(it)
+        }
+        requireActivity().registerReceiver(messageReceiver, IntentFilter(MESSAGES_DATA_UPDATED))
+    }
+
+    override fun onStop() {
+        super.onStop()
+        requireActivity().unregisterReceiver(messageReceiver)
```
Comment (shako-midelauri) [inline app/src/main/java/com/c0d3in3/chatapp/ui/ChatFragment.kt:70]: If you register receiver in onResume then you have to unregister it in the onPause, if you register it in onStart you have to unregister it in onStop. Generally you have to register/unregister in “opposite” lifecycles.

Comment (ted3x) [reply app/src/main/java/com/c0d3in3/chatapp/ui/ChatFragment.kt:70]: Changed

**Review by shako-midelauri — APPROVED** [review]

File: app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    private var messages = listOf<Message>()
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
+        val myView: View
+        val model = messages[position]
+        when (getItemViewType(position)) {
+            SELF_MESSAGE -> {
+                myView = LayoutInflater.from(parent!!.context)
+                    .inflate(R.layout.self_message_item, parent, false)
+                myView.selfTV.text = model.message
```
Comment (Edna001) [inline app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt:32]: apply_ით გადააკეთე

File: app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                myView.selfDateTV.text = model.messageTimestamp.toDate()
+            }
+            else -> {
+                myView = LayoutInflater.from(parent!!.context)
+                    .inflate(R.layout.not_self_message_item, parent, false)
+                myView.notSelfTV.text = model.message
```
Comment (Edna001) [inline app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt:38]: apply_ით გადააკეთე

File: app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                myView.notSelfDateTV.text = model.messageTimestamp.toDate()
+            }
+        }
+        return myView
+    }
+
+    override fun getItem(position: Int) = messages[position]
+
+    override fun getItemId(position: Int) = 0.toLong()
```
Comment (Edna001) [inline app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt:47]: რატო 0 და არა position.toLong()?

File: app/src/main/java/com/c0d3in3/chatapp/ui/ChatFragment.kt [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+import com.c0d3in3.chatapp.Constants.MESSAGE_MODEL_KEY
+import com.c0d3in3.chatapp.MessageReceiver
+
+import com.c0d3in3.chatapp.R
+import com.c0d3in3.chatapp.model.Message
+import com.c0d3in3.chatapp.model.MessageOwner
+import com.c0d3in3.chatapp.ui.adapter.ChatAdapter
+import kotlinx.android.synthetic.main.chat_fragment.*
+
+class ChatFragment(private val messageOwner: MessageOwner) : Fragment() {
```
Comment (Edna001) [inline app/src/main/java/com/c0d3in3/chatapp/ui/ChatFragment.kt:26]: ჩემი აზრით ჯობია ორი ფრაგმენტით გააკეთო

File: app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt (hunk 4 above)
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt:47]: Yes you are right. using position as itemId makes sense.

File: app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt (hunk 3 above)
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt:38]: Changed

File: app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt (hunk 2 above)
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/chatapp/ui/adapter/ChatAdapter.kt:32]: Changed

**Review by Edna001 — APPROVED** [review]

**Review by anarusadzeSpace — APPROVED** [review]

**Pull request merged**